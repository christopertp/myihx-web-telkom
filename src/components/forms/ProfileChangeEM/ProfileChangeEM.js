import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { Field } from 'redux-form';
import Button from '../../elements/Button';
import Spinner from '../../elements/Spinner';
import TextField from '../../fields/Text';
import styles from './styles.scoped.css';

export default function ProfileChangeEM(props) {
  const { handleSubmit, invalid } = props;
  const { isLoading } = useSelector(s => s.profile);
  const { page } = useParams();
  const isChangeEmail = page === 'change-email';
  const inputProps = {
    label: isChangeEmail ? 'Email Address' : 'Mobile Number',
    placeholder: isChangeEmail ? 'Enter valid email address' : 'Enter Phone Number',
    type: isChangeEmail ? 'email' : 'tel',
  };
  const fieldName = isChangeEmail ? 'email' : 'mobile';

  return (
    <form className={styles.root} onSubmit={handleSubmit}>
      <Field component={TextField} inputProps={inputProps} name={fieldName} />
      <Button disabled={invalid || isLoading.b} type="submit">
        {isLoading.b ? <Spinner /> : 'Next'}
      </Button>
    </form>
  );
}

ProfileChangeEM.defaultProps = {
  handleSubmit: () => {},
  invalid: false,
};

ProfileChangeEM.propTypes = {
  handleSubmit: PropTypes.func,
  invalid: PropTypes.bool,
};
