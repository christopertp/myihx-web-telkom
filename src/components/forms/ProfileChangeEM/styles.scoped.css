.root > button {
  margin-top: 1.5rem;
  width: 100%;
}

@media only screen and (max-width: 48em) {
  .root { margin-top: 2rem;}
  .root > button { margin-top: 5rem; }
}
