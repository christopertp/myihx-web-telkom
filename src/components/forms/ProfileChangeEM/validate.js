import { isEmail, isPhone } from '../../../utils/validation';

export default function validate(values) {
  const { email, mobile } = values;

  return {
    email: !isEmail(email) ? 'Mohon cantumkan email yang aktif' : '',
    mobile: !isPhone(mobile) ? 'Mohon cantumkan nomor ponsel yang aktif' : '',
  };
}
