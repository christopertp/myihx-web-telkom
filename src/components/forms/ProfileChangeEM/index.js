import { reduxForm } from 'redux-form';
import ProfileChangeEM from './ProfileChangeEM';
import validate from './validate';

export default reduxForm({
  initialValues: {
    email: '',
    mobile: '',
  },
  form: 'profileChangeEM',
  validate,
})(ProfileChangeEM);
