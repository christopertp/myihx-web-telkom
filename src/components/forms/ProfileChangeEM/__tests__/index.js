import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import ShallowRenderer from 'react-test-renderer/shallow';
import Spinner from '../../../elements/Spinner';
import ProfileChangeEM from '../ProfileChangeEM';

useSelector.mockImplementation(fn => {
  fn({});
  return { isLoading: false, };
});

describe('src/components/forms/ProfileChangeEM', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ProfileChangeEM />);
    expect(tree).toMatchSnapshot();

    useSelector.mockImplementationOnce(() => ({ isLoading: { b: true }, }));
    ProfileChangeEM.defaultProps.handleSubmit();
    const result = ProfileChangeEM({
      ...ProfileChangeEM.defaultProps,
    });
    const button = result.props.children[1];
    expect(button.props.children.type).toBe(Spinner);

    useParams.mockImplementationOnce(() => ({ page: 'change-email' }));
    useSelector.mockImplementationOnce(() => ({ isLoading: { b: false }, }));
    const result2 = ProfileChangeEM({
      ...ProfileChangeEM.defaultProps,
    });
    const field = result2.props.children[0];
    expect(field.props.name).toBe('email');
  });
});
