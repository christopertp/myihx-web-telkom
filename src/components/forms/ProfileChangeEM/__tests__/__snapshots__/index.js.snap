// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/forms/ProfileChangeEM render 1`] = `
<form
  onSubmit={[Function]}
>
  <Field
    component={[Function]}
    inputProps={
      Object {
        "label": "Mobile Number",
        "placeholder": "Enter Phone Number",
        "type": "tel",
      }
    }
    name="mobile"
  />
  <Button
    className=""
    disabled={false}
    onClick={null}
    to=""
    type="submit"
    variant="red"
  >
    Next
  </Button>
</form>
`;
